back/src/api/declaration/putDeclaration/validator.ts

```diff
@@ -1,81 +1,88 @@
 import { Alpha2Code, getAlpha2Codes } from 'i18n-iso-countries';
-import { buildValidationMiddleware, IRequestValidatorSchema } from '../../../core/middlewares';
-import { validator } from '../../../core/validator';
+import { z } from 'zod';
+import { buildValidationMiddleware } from '../../../core/middlewares';
 import { AuthorType } from '../../common/enums/author.enum';
-import { meansOfTransport, MeansOfTransport } from '../../common/enums/meansOfTransport.enum';
+import { MeansOfTransport } from '../../common/enums/meansOfTransport.enum';
+import { parseNumber } from '../../../utils/zodParser';
+import { refineMeanOfTransport } from '../getSimulation/validator';
 
-export interface PutDeclarationRequest {
-  params: {
-    declarationId: string;
-  };
-  body: {
-    shoppingProducts: {
-      id?: string;
-      customName?: string;
-      customId: string;
-      originalValue: number;
-      currency?: string;
-    }[];
-    border: boolean;
-    age: number;
-    country: Alpha2Code;
-    meanOfTransport?: MeansOfTransport;
-    authorType: AuthorType;
-    declarantAddressStreet: string;
-    declarantAddressPostalCode: string;
-    declarantAddressCity: string;
-    declarantEmail: string;
-    declarantPhoneNumber: string | null;
-    declarantFirstName: string;
-    declarantLastName: string;
-  };
-}
-
-export const putDeclarationValidator: IRequestValidatorSchema = {
-  params: validator.object({
-    declarationId: validator.string().uuid().required(),
+export const putDeclarationValidator = z.object({
+  params: z.object({
+    declarationId: z
+      .string({
+        required_error: "L'id de la déclaration est requis",
+      })
+      .uuid(),
   }),
-  body: validator.object({
-    shoppingProducts: validator
-      .array()
-      .items(
-        validator.object({
-          id: validator.string().uuid(),
-          customName: validator.string().allow(''),
-          customId: validator.string().uuid().required(),
-          originalValue: validator.number().min(0).invalid(0).required(),
-          currency: validator.string().length(3).default('EUR'),
+  body: z
+    .object({
+      shoppingProducts: z.array(
+        z.object({
+          id: z.string().uuid().optional(),
+          customName: z.string().optional().default(''),
+          customId: z
+            .string({
+              required_error: "L'identifiant du produit est requis",
+            })
+            .uuid(),
+          originalValue: parseNumber(z.number().min(0)),
```

Comment:
ici il faut conserver le fait que 0 n'est pas accepté

```suggestion
          originalValue: parseNumber(z.number().gt(0)),
```